cores/esp8266/Arduino.h

```diff
@@ -33,6 +33,25 @@ extern "C" {
 #include <string.h>
 #include <math.h>
 
+#if defined(__cplusplus) && !defined(__cpp_exceptions)
+// this is a hack working with gcc4.8
+// Overload operator new so it can return nullptr instead of throwing an
+// exception without (std::nothrow)`
+// Requirement (or the constructor is called even on nullptr)
+// - "noexcept" is needed
+// - "#include <new>" must not be done *before* these definitions
+#include <bits/c++config.h>
+extern "C++" inline void* operator new (std::size_t size) noexcept
+{
+    return malloc(size);
```

Comment:
Please check out the old new() and new[]() override, it sets the last-failed-alloc address which this needs to do as well:
https://github.com/esp8266/Arduino/blob/bb28d4a3927758b0b345579792cdeebbc9e2e6a3/cores/esp8266/abi.cpp#L31-L49